arq/worker.py

```diff
@@ -295,13 +298,20 @@ def run(self) -> None:
             job_ids = await self.pool.zrangebyscore(
                 self.queue_name, offset=self._queue_read_offset, count=count, max=now
             )
+            self.abort_tasks = [
```

Comment:
Can we have a switch (default off) for whether we need to abort tasks, so this step can be omitted for those who don't need it. (this is a performance critical part of arq and `keys('...*')` is fairly slow I think.

Does `abort_tasks` need to be `self.abort_tasks`? I don't see why. Also gain, use f-strings.

Also, use a set not a list.

Reply:
also, see [this](https://redis.io/commands/keys) warning, could we use, scan instead?

Reply:
Sure, i changed a bit the code to use it more efficiently (I think). The results is something like that:

```python
    async def _abort_jobs(self):
        todel, cur = [], b'0'
        lk = len(abort_key_prefix)
        pool, tasks = self.pool, self.tasks
        while cur:
            cur, keys = await pool.scan(cur, match=f'{abort_key_prefix}*')
            for job_key in keys:
                job_id = job_key[lk:]
                if job_id not in tasks:
                    todel.append(job_key)
                    continue
                tasks[job_id].cancel()
                self._aborting_taks.add(job_id)
        if todel:
            await pool.delete(*todel)

    async def _poll_iteration(self):
        count = self.queue_read_limit
        if self.burst and self.max_burst_jobs >= 0:
            burst_jobs_remaining = self.max_burst_jobs - self._jobs_started()
            if burst_jobs_remaining < 1:
                return
            count = min(burst_jobs_remaining, count)

        async with self.sem:  # don't bother with zrangebyscore until we have "space" to run the jobs
            now = timestamp_ms()
            job_ids = await self.pool.zrangebyscore(
                self.queue_name, offset=self._queue_read_offset, count=count, max=now
            )
            if self.abort_jobs:
                await self._abort_jobs()
        await self.run_jobs(job_ids)

        for t in list(self.tasks.values()):
            if t.done():
                del self.tasks[job_id]
                # required to make sure errors in run_job get propagated
                t.result()

        await self.heart_beat()
```
What do you think?